### ISSUE TITLE
jobs Dependency

### ISSUE THREAD

**Issue by yxhmjl** [kubernetes/kubernetes#115370]
### What would you like to be added?

<img width="601" alt="image" src="https://user-images.githubusercontent.com/48946210/215252145-80c24753-c4d5-4a73-9513-3b28721a67ca.png">

### Why is this needed?

Hi  
Soamtimes our tasks have a stepwise requirement, we need to wait until some of the jobs in step1 are completed before we can execute the jobs in step2, and all the tasks in step2 are completed before we can execute the job in step3.
I don't know how to do in kubernetes.
thank you

**Comment by kannon92** [comment]
One project that would perfectly for this use case is https://argoproj.github.io/argo-workflows.  

I’m not sure workflow definitions is in scope for job aspirin so I’d suggest looking at Argo-workflows

**Comment by pacoxu** [comment]
https://github.com/kubernetes-sigs/kueue and https://github.com/kubernetes-sigs/kube-batch may help.

**Comment by alculquicondor** [comment]
This is out-of-scope for kubernetes at the moment https://github.com/kubernetes/community/blob/master/wg-batch/charter.md#out-of-scope

There are multiple third-party controllers that support this on top of Kubernetes, like:
- Argo workflows
- Tekton
- Nextflow

And I'm probably missing a lot :)

/close
